### ISSUE TITLE
compile error on 0.34.1

### ISSUE THREAD

**Issue by happysalada** [vectordotdev/vector#19191]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

compile error on latest vector 0.34.1

```
error: call to `.clone()` on a reference in this situation does nothing
   --> src/convert_config.rs:160:93
    |
160 |                       .unwrap_or_else(|| panic!("Failed to get file_name for {ent>
    |  ________________________________________________________________________________>
161 | |                     .clone();
    | |____________________________^ help: remove this redundant call
    |
    = note: the type `std::ffi::OsStr` does not implement `Clone`, so calling `clone` o>
note: the lint level is defined here
   --> src/lib.rs:7:9
    |
7   | #![deny(warnings)]
    |         ^^^^^^^^
    = note: `#[deny(noop_method_call)]` implied by `#[deny(warnings)]`

error: could not compile `vector` (lib) due to previous error
```

### Configuration

```text
rust 1.73.0
```

### Version

0.34.1

### Additional Context

getting this in the nixos update.

**Comment by Salathaniel** [comment]
https://gptzero.me/?via=salathaniel

**Comment by jpds** [comment]
Already fixed by https://github.com/vectordotdev/vector/pull/19075 (this is an issue with `.0`, not `.1`).

**Comment by StephenWakely** [comment]
This shouldn't be a problem if you are compiling with `rustc` version 1.72.1 as per [`rust-toolchain.toml`](https://github.com/vectordotdev/vector/blob/master/rust-toolchain.toml).

As @jpds the problem has been fixed on master by #19075.

**Comment by happysalada** [comment]
Thank you both for your prompt responses !
